src/main/java/in/twizmwaz/cardinal/module/modules/wools/WoolObjective.java

```diff
@@ -145,11 +145,7 @@ public void onWoolPickup(InventoryClickEvent event) {
                         this.touched = true;
                         if (touchMessage) {
                             double newProx;
-                            if (location != null) {
-                                newProx = location.distance(place.getVector());
-                            } else {
-                                newProx = player.getLocation().toVector().distance(place.getVector());
-                            }
+                            newProx = player.getLocation().toVector().distance(place.getVector());
```

Comment:
@iEli2tyree011 Made some changes, now if the players pick up the wool clicking it in a chest (that should be the first touch) proximity will use location XML to calculate the distance. In the case that the wool is in the floor or dropped, proximity will use the distance between player-monument to calculate the vector. That's what do you want?